vllm/sampling_params.py

```diff
@@ -426,6 +426,9 @@ def _verify_args(self) -> None:
         if not 0.0 <= self.min_p <= 1.0:
             raise ValueError("min_p must be in [0, 1], got "
                              f"{self.min_p}.")
+        if self.min_p > 0.0 and self.temperature == 0.0:
+            raise ValueError("min_p > 0.0 requires random sampling "
+                             "but temperature == 0.0")
```

Comment:
I'm not sure that we want to add this. I agree this is good validation to have but:
- If we do it for this we should do it for all such parameters (top_p, top_k, etc.)
- I think that adding this may break a lot of existing callers who happen to be setting these together (even though the min_p is having no effect for them in that case)

I think this should be dealt with in a separate PR (if at all) ... we'd likely need to control the behaviour with an env var ("strict validation of non-greedy sampling params")